lib/rucio/daemons/reaper/reaper.py

```diff
@@ -712,10 +711,10 @@ def run(
         raise DatabaseException('Database was not updated, daemon won\'t start')
 
     logging.log(logging.INFO, 'main: starting processes')
-    rses_to_process = get_rses_to_process(rses, include_rses, exclude_rses, vos)
-    if not rses_to_process:
-        logging.log(logging.ERROR, 'Reaper: No RSEs found. Exiting.')
-        return
+
+    while not (rses_to_process := get_rses_to_process(rses, include_rses, exclude_rses, vos)):
+        logging.warning(f"Reaper: No RSEs found. Sleeping for {sleep_time} seconds and retrying.")
+        time.sleep(sleep_time)
```

Comment:
I would have approached this differently. 

In this scope, `rses_to_process` is used exclusively for one logging entry. It has no other functional purpose. However, this entry is misleading: it outputs the list of RSEs at the time the daemon starts, but that may change during its operation. In addition, the actual list of RSEs is calculated in `run_once()`. Consequently, I would remove this entirely.

Note that using f-strings and other formatting constructs in calls to `logging.log()` is an [anti-pattern](https://docs.astral.sh/ruff/rules/logging-f-string/) which we’re actively moving away from.

Reply:
Rewrote in https://github.com/rucio/rucio/pull/7472/commits/0cf34c28ea38c19f3d22827bb1c612f63d4225d9